fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -802,13 +802,27 @@ private void addScanRangeLocations(Partition partition,
             paloRange.setVersion(visibleVersionStr);
             paloRange.setVersionHash("");
             paloRange.setTabletId(tabletId);
+            if (ConnectContext.get().isTxnModel()) {
+                paloRange.setSubTxnIds(
+                        ConnectContext.get().getTxnEntry().getTabletSubTxnIds(olapTable.getId(), tablet));
+                LOG.info("table={}, partition={}, tablet={}, sub txn ids={}", olapTable.getId(), partition.getId(),
+                        tablet.getId(), paloRange.getSubTxnIds());
+            }
 
```

Comment:
We should assure that select in txn are forward to master too.

Reply:
the observer also knows the sub txn info